cmd/create/operatorroles/by_clusterkey.go

```diff
@@ -27,68 +27,70 @@ func handleOperatorRoleCreationByClusterKey(r *rosa.Runtime, env string,
 		r.Reporter.Errorf("Cluster '%s' is not an STS cluster.", clusterKey)
 		os.Exit(1)
 	}
-	// Check to see if IAM operator roles have already created
-	missingRoles, err := validateOperatorRoles(r, cluster)
-	if err != nil {
-		if strings.Contains(err.Error(), "AccessDenied") {
-			r.Reporter.Debugf("Failed to verify if operator roles exist: %s", err)
-		} else {
-			r.Reporter.Errorf("Failed to verify if operator roles exist: %s", err)
-			os.Exit(1)
-		}
-	}
 
-	if len(missingRoles) == 0 {
-		if ocm.IsOidcConfigReusable(cluster) {
-			err := validateOperatorRolesMatchOidcProvider(r, cluster)
-			if err != nil {
-				return err
-			}
-		}
+	hostedCPPolicies := aws.IsHostedCPManagedPolicies(cluster)
 
-		if !args.forcePolicyCreation {
-			r.Reporter.Infof("Operator Roles already exists")
-			return nil
-		}
-	}
-	roleName, err := aws.GetInstallerAccountRoleName(cluster)
+	operatorRolePolicyPrefix, err := aws.GetOperatorRolePolicyPrefixFromCluster(cluster, r.AWSClient)
 	if err != nil {
-		r.Reporter.Errorf("Expected parsing role account role '%s': %v", cluster.AWS().STS().RoleARN(), err)
-		os.Exit(1)
+		r.Reporter.Errorf("%s", err)
 	}
 
-	path, err := aws.GetPathFromAccountRole(cluster, aws.AccountRoles[aws.InstallerAccountRole].Name)
+	credRequests, err := r.OCMClient.GetCredRequests(cluster.Hypershift().Enabled())
 	if err != nil {
-		r.Reporter.Errorf("Expected a valid path for '%s': %v", cluster.AWS().STS().RoleARN(), err)
+		r.Reporter.Errorf("Error getting operator credential request from OCM %s", err)
 		os.Exit(1)
 	}
-	if path != "" && !output.HasFlag() && r.Reporter.IsTerminal() {
-		r.Reporter.Infof("ARN path '%s' detected in installer role '%s'. "+
-			"This ARN path will be used for subsequent created operator roles and policies.",
-			path, cluster.AWS().STS().RoleARN())
-	}
-	var accountRoleVersion string
 
 	managedPolicies := cluster.AWS().STS().ManagedPolicies()
 	if args.forcePolicyCreation && managedPolicies {
 		r.Reporter.Warnf("Forcing creation of policies only works for unmanaged policies")
 		os.Exit(1)
 	}
-	credRequests, err := r.OCMClient.GetCredRequests(cluster.Hypershift().Enabled())
-	if err != nil {
-		r.Reporter.Errorf("Error getting operator credential request from OCM %s", err)
-		os.Exit(1)
-	}
-
-	operatorRolePolicyPrefix, err := aws.GetOperatorRolePolicyPrefixFromCluster(cluster, r.AWSClient)
-	if err != nil {
-		r.Reporter.Errorf("%s", err)
-	}
-
-	hostedCPPolicies := aws.IsHostedCPManagedPolicies(cluster)
 
 	switch mode {
 	case aws.ModeAuto:
+		// Check to see if IAM operator roles have already created
+		missingRoles, err := validateOperatorRoles(r, cluster)
+		if err != nil {
+			if strings.Contains(err.Error(), "AccessDenied") {
+				r.Reporter.Debugf("Failed to verify if operator roles exist: %s", err)
+			} else {
+				r.Reporter.Errorf("Failed to verify if operator roles exist: %s", err)
```

Comment:
Single quotes on %s missing

Reply:
i can add that, i didnt make any of this code though so dont blame me :laughing:

Reply:
Yeah I understand haha , just to keep code similarity across the repo :)